### PR TITLE
kv tests

### PR DESCRIPTION
**Pull request by saxena52**
I tested the following commands for key value: kv set/get, clean, del, expire/ttl, and list. I will later add test for getMany, llen, lrange, rpush, and setMany.

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -1,12 +1,13049 @@
 {
   "name": "nimbella-cli",
   "version": "1.16.0",
-  "lockfileVersion": 1,
+  "lockfileVersion": 2,
```
Comment (rabbah) [inline package-lock.json:4]: @saxena52 back out this file from the PR - it shouldn't be needed for the new tests.

File: tests/keyvalue/test.bats [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+setup_file() { # kv delete being tested here
+	$NIM key-value delete $KEY_NAME
+}
+
+teardown_file() { # kv delete also being tested here
+	$NIM key-value delete $KEY_NAME
+}
+
+@test "kv set/get" { # kv set/get test
+	testValue=$(date)
```
Comment (rabbah) [inline tests/keyvalue/test.bats:18]:
```suggestion
  testValue=$(date)
```

Replace tabs with white space.
@jthomas I guess we don't run scan code on these files - is there some convenient way to enforce formatting for bats?

File: tests/keyvalue/test.bats [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+}
+
+@test "kv list" { # kv list test
+
+  run $NIM key-value set ANOTHER_KEY "ANOTHER_VALUE" # adding another item to populate list
+
+  listString="ANOTHER_KEY"$'\n'"${KEY_NAME}"
+
+  run $NIM key-value list
+	assert_success
```
Comment (rabbah) [inline tests/keyvalue/test.bats:37]:
```suggestion
  assert_success
```

File: tests/keyvalue/test.bats [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  assert_output "${listString}"
+	
+}
+
+@test "kv expire/ttl" { # kv expire/ttl test
+
+  run $NIM key-value expire $KEY_NAME "3" # expire time set for 3 seconds
+	assert_success
```
Comment (rabbah) [inline tests/keyvalue/test.bats:45]:
```suggestion
  assert_success
```

File: tests/keyvalue/test.bats [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+	assert_output "1"
```
Comment (rabbah) [inline tests/keyvalue/test.bats:46]:
```suggestion
  assert_output "1"
```

File: tests/keyvalue/test.bats [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+
+  run $NIM key-value ttl $KEY_NAME # checking remaining time of key
+	assert_success
```
Comment (rabbah) [inline tests/keyvalue/test.bats:49]:
```suggestion
  assert_success
```

File: tests/keyvalue/test.bats [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+  [ "$status" -le "3" ]
+
+  sleep 3 # pausing script for 3 seconds
+
+  run $NIM key-value get $KEY_NAME # checking to see if key still exists
+	assert_success
```
Comment (rabbah) [inline tests/keyvalue/test.bats:55]:
```suggestion
  assert_success
```

File: tests/keyvalue/test.bats [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+	assert_output "null"
```
Comment (rabbah) [inline tests/keyvalue/test.bats:56]:
```suggestion
  assert_output "null"
```

File: tests/keyvalue/test.bats [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+	
+}
+
+@test "kv clean" { # kv clean test
+
+  run $NIM key-value clean --force
+	assert_success
```
Comment (rabbah) [inline tests/keyvalue/test.bats:63]:
```suggestion
  assert_success
```

File: tests/keyvalue/test.bats [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+	assert_output "all content cleared"
```
Comment (rabbah) [inline tests/keyvalue/test.bats:64]:
```suggestion
  assert_output "all content cleared"
```

File: tests/keyvalue/test.bats [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+
+  run $NIM key-value list
+	assert_success
```
Comment (rabbah) [inline tests/keyvalue/test.bats:67]:
```suggestion
  assert_success
```

**Review by rabbah — COMMENTED** [review]
🎉 congrats on your first PR to the project.

LGTM modulo some formatting nits.

File: tests/keyvalue/test.bats (hunk 2 above)
Comment (jthomas) [reply tests/keyvalue/test.bats:18]: @rabbah Some Googling leads me to.. https://github.com/mvdan/sh I wonder if this would work?

**Comment by jthomas** [comment]
This is wonderful - thanks for contributing. It all looks really sensible. I concur with @rabbah minor nitpicks but apart from that - ready to merge!

**Review by saxena52 — COMMENTED** [review]
Have made all necessary changes with spacing, and package-lock.json should be reverted back to how it originally was.

**Review by rabbah — APPROVED** [review]
LGTM.

**Pull request merged**